src/System Application/App/Azure Key Vault/src/AzureKeyVaultImpl.Codeunit.al

```diff
@@ -30,18 +30,25 @@ codeunit 2202 "Azure Key Vault Impl."
         IsKeyVaultClientInitialized: Boolean;
         AzureKeyVaultTxt: Label 'Azure Key Vault', Locked = true;
         CertificateInfoTxt: Label 'Successfully constructed certificate from secret %1. Certificate thumbprint %2', Locked = true;
+        MissingSecretErr: Label '%1 is either missing or empty', Comment = '%1 = Secret Name.';
 
     [NonDebuggable]
     procedure GetAzureKeyVaultSecret(SecretName: Text; var Secret: Text)
     begin
         // Gets the secret as a Text from the key vault, given a SecretName.
         Secret := GetSecretFromClient(SecretName);
+
+        if Secret = '' then
+            Error(MissingSecretErr, SecretName);
```

Comment:
This error will break features that rely on an empty string being returned in certain cases, such as OpenAI which assumes the secret will be empty if disabled and then it will log a nice error to telemetry and show a nice error to the user rather than the secret being missing:
![image](https://github.com/user-attachments/assets/b5edacf9-ca20-4281-a2d6-5bb0add5239c)